src/main/webapp/app/exercises/shared/result/result-progress-bar/result-progress-bar.component.ts

```diff
@@ -0,0 +1,113 @@
+import { Component, OnDestroy, effect, input } from '@angular/core';
+import { CommonModule } from '@angular/common';
+import { ArtemisSharedCommonModule } from 'app/shared/shared-common.module';
+import { faCircleNotch } from '@fortawesome/free-solid-svg-icons';
+
+@Component({
+    selector: 'jhi-result-progress-bar',
+    standalone: true,
+    imports: [CommonModule, ArtemisSharedCommonModule],
+    templateUrl: './result-progress-bar.component.html',
+    styleUrl: './result-progress-bar.component.scss',
+})
+export class ResultProgressBarComponent implements OnDestroy {
+    estimatedRemaining = input<number>(0);
+    estimatedDuration = input<number>(0);
+    isBuilding = input<boolean>();
+    isQueued = input<boolean>();
+    showBorder = input<boolean>(false);
+
+    isQueueProgressBarAnimated: boolean;
+    queueProgressBarOpacity: number;
+    queueProgressBarValue: number;
+
+    isBuildProgressBarAnimated: boolean;
+    buildProgressBarOpacity: number;
+    buildProgressBarValue: number;
+
+    estimatedDurationInterval: ReturnType<typeof setInterval> | undefined;
+
+    faCircleNotch = faCircleNotch;
```

Comment:
We also usually define the constants at the top
```suggestion
    protected readonly faCircleNotch = faCircleNotch;
```